dhall/src/Dhall/Parser/Expression.hs

```diff
@@ -271,22 +271,11 @@ parsers embedded = Parsers {..}
                         _ -> return (Annot a b)
 
             let alternative4C = do
-                    case shallowDenote a of
-                        Equivalent{}   -> empty
-                        ImportAlt{}    -> empty
-                        BoolOr{}       -> empty
-                        NaturalPlus{}  -> empty
-                        TextAppend{}   -> empty
-                        ListAppend{}   -> empty
-                        BoolAnd{}      -> empty
-                        Combine{}      -> empty
-                        Prefer{}       -> empty
-                        CombineTypes{} -> empty
-                        NaturalTimes{} -> empty
-                        BoolEQ{}       -> empty
-                        BoolNE{}       -> empty
-                        App{}          -> empty
-                        _              -> return ()
+                    -- Ensure that what we've parsed so far is an import expression.
+                    case a0Type of
+                        ApplicationExpr               -> empty
+                        ImportExpr | eqExprBySrc a0 a -> return ()
```

Comment:
What's the reason for the `eqExprBySrc` check here?

Reply:
If I remove it, we'd parse a `with` expression even when `a` is an operator expression.

This causes [`tests/parser/failure/unit/WithPrecedence1.dhall`](https://github.com/dhall-lang/dhall-lang/blob/0c899ab31fdb857f535fe9f5b37f981271824a65/tests/parser/failure/unit/WithPrecedence1.dhall) to fail.